api/caseshare/real-api.ts

```diff
@@ -0,0 +1,160 @@
+import { SharedCase } from '@hmcts/rpx-xui-common-lib/lib/models/case-share.model'
+import { Request, Response } from 'express'
+import { handleGet, handlePost } from '../common/crudService'
+import { getConfigValue } from '../configuration'
+import { CASE_SHARE_PERMISSIONS, SERVICES_MCA_PROXY_API_PATH, SERVICES_RD_PROFESSIONAL_API_PATH } from '../configuration/references'
+import * as log4jui from '../lib/log4jui'
+import { JUILogger } from '../lib/models'
+import { toCaseAssigneeMappingModel } from './dtos/case-user-dto'
+import { ccdToUserDetails, prdToUserDetails } from './dtos/user-dto'
+import { CaseAssigneeMappingModel } from './models/case-assignee-mapping.model'
+import { CCDRawCaseUserModel } from './models/ccd-raw-case-user.model'
+
+const logger: JUILogger = log4jui.getLogger('real-api')
+const prdUrl: string = getConfigValue(SERVICES_RD_PROFESSIONAL_API_PATH)
+const ccdUrl: string = getConfigValue(SERVICES_MCA_PROXY_API_PATH)
+
+export async function getUsers(req: Request, res: Response) {
+  try {
+    const path = `${prdUrl}/refdata/external/v1/organisations/users?returnRoles=true&status=active`
+    const {status, data}: {status: number, data: any} = await handleGet(path, req)
+    const permissions = CASE_SHARE_PERMISSIONS.split(',')
+    const users = [...data.users]
+                  .filter(user => user && user.roles && user.roles.some(role => permissions.includes(role)))
+                  .map(user => prdToUserDetails(user))
+    return res.status(status).send(users)
+  } catch (error) {
+    return res.status(error.status).send({
+      errorMessage: error.data,
+      errorStatusText: error.statusText,
+    })
+  }
+}
+
+export async function getCases(req: Request, res: Response) {
+  try {
+    const caseIds = req.query.case_ids
+    const path = `${ccdUrl}/case-assignments?case_ids=${caseIds}`
+    const {status, data}: {status: number, data: any} = await handleGet(path, req)
+    const caseUsers: CCDRawCaseUserModel[] = [...data.case_assignments]
+    const sharedCases: SharedCase[] = []
+    for (const caseUser of caseUsers) {
```

Comment:
I would be tempted to move this for of loop out into a separate function, as it makes it easily testable, and then you don't have the getCases function doing two things, ie. a function should do one thing.

Reply:
<-- Phil to look at .